Forbidding local npm installs

luisherranz opened the pull request:
<!--
Thanks for your pull request 😊. Note that not following the template might
result in your issue being closed.
-->

<!--
Please make sure you're familiar with and follow the instructions in the
contributing guidelines found in the
https://docs.frontity.org/contributing/code-contributions.
-->

<!--
If you're new to contributing to open source projects, you might find this free
video course helpful: http://kcd.im/pull-request
-->

<!--
Please fill out the information below to expedite the review and (hopefully)
merge of your pull request!
-->

**What**:

<!-- What changes are being made? (What feature/bug is being fixed here?) -->

Forbid running local `npm install`, inside of the packages.

**Why**:

<!-- Why are these changes necessary? -->

Because it is not straightforward that people should only use `npm install` in the root and the `e2e` folder, and not in the packages or projects.

**How**:

<!-- How were these changes implemented? -->

I added a `preinstall` script to all our packages that echos a message and stops the install.

I add to also use lerna's `--ignore-scripts` option and run `lerna run build` after the install to make lerna not to run the `preinstall` commands.

**Tasks**:

<!-- Have you done all of these things?  -->

<!-- To check an item, place an "x" in the box like so: "- [x] Unit tests" -->

<!-- Move any unrelated task to the Unrelated tasks section below. -->

- [x] Package.json scripts

<!-- Changesets are necessary if your changes should release any packages.
Run `npx changeset` to create a changeset.
More info at https://docs.frontity.org/contributing/code-contribution-guide#what-is-a-changeset -->

**Unrelated Tasks**

<!-- ignore-task-list-start -->

- [ ] Code
- [ ] TSDocs
- [ ] TypeScript
- [ ] Unit tests
- [ ] End to end tests
- [ ] TypeScript tests
- [ ] Update starter themes
- [ ] Update other packages
- [ ] Open an issue for this feature in the [docs repo](https://github.com/frontity/docs/wiki/Code-Releases)
- [ ] Update community discussions
- [ ] Add a changeset (with link to its [Feature Discussion](https://community.frontity.org/c/33) if it exists)
<!-- ignore-task-list-end -->

**Additional Comments**

<!-- Feel free to add any additional comments. -->

I am not 100% sure about the approach. Let's just try and see. No problem as this is only for the Frontity Core team 🙂

luisherranz committed:
Add forbidden install to e2e project

luisherranz committed:
Add forbidden install to packages

DAreRodz reviewed (approved):
I think is OK so I'll approve this. :+1: 

BTW, did you notice that `lerna run build` is building also the example packages and the e2e project? For me, that takes a quite long time (not so long, anyway).

Maybe we could filter the packages that run the `build` script? The bad news is that there's no way to [filter packages by path](https://github.com/lerna/lerna/issues/1840#issuecomment-469728168) using `lerna`. We would have to look for a different solution.

luisherranz committed:
Lerna: Run prepublish instead of build

Because that is what lerna was running, and the examples and
project folders also have a build script that we don't want to run.

luisherranz commented:
You are absolutely right, good catch. I have changed it to `lerna run prepublish`, which was what lerna was doing anyway.

luisherranz committed:
Merge branch 'dev' into forbing-local-npm-installs

DAreRodz commented:
Using `prepublish` is definitely better. :blush: Feel free to merge this, @luisherranz.

luisherranz committed:
Merge branch 'dev' into forbing-local-npm-installs

luisherranz committed:
Update author tests snapshot

luisherranz committed:
Merge remote-tracking branch 'origin/dev' into forbing-local-npm-installs
